pkg/phlaredb/block/list.go

```diff
@@ -63,8 +63,12 @@ func IterBlockMetas(ctx context.Context, bkt phlareobj.Bucket, from, to time.Tim
 		for _, id := range ids {
 			id := id
 			g.Go(func() error {
-				r, err := bkt.Get(ctx, id+MetaFilename)
+				r, err := bkt.Get(ctx, path.Join(id, MetaFilename))
 				if err != nil {
+					if bkt.IsObjNotFoundErr(err) {
+						level.Info(util.Logger).Log("msg", "block meta.json not found while iterating", "id", id)
```

Comment:
Maybe that makes it clearer, that this is not an error:

```suggestion
						level.Info(util.Logger).Log("msg", "skipping block as meta.json not found", "id", id)
```